lib/vehicle_factory.rb

```diff
@@ -0,0 +1,24 @@
+class VehicleFactory
+  attr_reader :vehicles
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(state_registrations)
+    state_registrations.map do |vehicle|
+      vehicle_details = {
+        vin: vehicle[:vin_1_10],
+        year: vehicle[:model_year],
+        make: vehicle[:make],
+        model: vehicle[:model],
+        engine: :ev,
+        registration_date: nil
+      }
+      
+      @vehicles << vehicle = Vehicle.new(vehicle_details)
```

Comment:
```suggestion
      @vehicles << Vehicle.new(vehicle_details)
```
You can actually go ahead and just shovel the vehicle in directly. No need to assign it to a variable if we aren't going to use the variable for anything.